price_service/server/src/promClient.ts

```diff
@@ -51,6 +51,49 @@ export class PromClient {
     }
   });
 
+  private priceIdToPriceUpdatePublishTimeGapHistogram = new Map<
+    string,
+    client.Histogram<string>
+  >();
```

Comment:
Is there any reason that you haven't used the `priceId` as a label and are one metric per each?